### PR TITLE
Use spectral attribute removal for normalization

### PR DESCRIPTION
**Pull request by norabelrose**
Too tired to explain what this is right now, will add a description tomorrow

### CODE REVIEW

**Comment by norabelrose** [comment]
Some bizarre nondeterministic test failure occurring right now

File: elk/extraction/extraction.py [hunk 1]
```diff
@@ -268,35 +267,27 @@ def extract_hiddens(
...
-                    log_p = outputs.logits[..., -answer_len:, :].log_softmax(dim=-1)
-                    tokens = answer[..., None]
-                    lm_logits[i, j] = log_p.gather(-1, tokens).mean()
-
-                elif isinstance(outputs, Seq2SeqLMOutput):
-                    # The cross entropy loss is averaged over tokens, so we need to
-                    # multiply by the length to get the total log probability.
-                    # length = encoding.labels.shape[-1]
-                    lm_logits[i, j] = -assert_type(Tensor, outputs.loss)  #  * length
+                    lm_logits[i, j] = -assert_type(Tensor, outputs.loss)
```
[the remark this answers is not in the document]
Comment (norabelrose) [reply elk/extraction/extraction.py:290]: We actually were already doing that before, since we used `.mean()`.

Somewhat embarrassingly, before we were not shifting the token IDs correctly when computing the log probability. Instead of trying to do this myself, I decided to make the code cleaner by relying on HuggingFace to handle the shifting and masking for us, taking advantage of the fact that `-100` is a special mask value.

At some point we should probably log both the length-normalized and non-length-normalized log probs

[the remark this answers is not in the document]
Comment (AlexTMallen) [reply elk/extraction/extraction.py:290]: Oh I wasn't aware that we switched to length-normalization before
